docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
 # Developer Guide
```

Comment:
Perhaps some of the images and diagrams are in lower resolution, maybe export to a higher resolution image instead?

Reply:
While it is a small issue, it is evident that the diagrams in this are drawn on different platform and have different colour schemes. Will it be better to fix on one colour scheme and one method to export the diagrams?